src/com/palmergames/bukkit/TownyChat/listener/EssentialsDiscordHookListener.java

```diff
@@ -0,0 +1,44 @@
+package com.palmergames.bukkit.TownyChat.listener;
+
+import com.palmergames.bukkit.TownyChat.Chat;
+import com.palmergames.bukkit.TownyChat.channels.Channel;
+import com.palmergames.bukkit.TownyChat.channels.channelTypes;
+import net.essentialsx.api.v2.events.discord.DiscordChatMessageEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EssentialsDiscordHookListener implements Listener {
+  private final Chat plugin;
+
+  public EssentialsDiscordHookListener(Chat plugin) {
+    this.plugin = plugin;
+  }
+
+  @EventHandler()
+  public void onDiscordChat(DiscordChatMessageEvent event) {
+    String directChat = plugin.getTownyPlayerListener().directedChat.get(event.getPlayer());
+    if (directChat != null) {
+      plugin.getTownyPlayerListener().directedChat.remove(event.getPlayer());
+      if (event.isCancelled()) {
+        return;
+      }
+      Channel channel = plugin.getChannelsHandler().getChannel(event.getPlayer(), directChat);
+      if (channel != null) {
+        event.setCancelled(!channel.isHooked() || channel.getType() != channelTypes.GLOBAL || channel.getRange() != -1);
```

Comment:
I think the 3 `!channel.isHooked()` requirements can be dropped, as that would just lead to more unnecessary setup required by the EssDiscord users. Testing that its not global and not multi-world alone should be good.